core/jvm/src/test/scala/fs2/ResourceCompilationSpec.scala

```diff
@@ -0,0 +1,95 @@
+package fs2
+
+import cats.implicits._
+import cats.effect.{IO, Resource}
+import cats.effect.concurrent.{Deferred, Ref}
+import scala.concurrent.duration._
+
+class ResourceCompilationSpec extends AsyncFs2Spec {
+  // NOTE: still doesn't work. There are Release nodes being inserted, whose finaliser gets called anyway.
```

Comment:
This comment is out of date right? The disabling of release if parent is defined fixes this issue?

Reply:
Yes, sorry, there is cleanup work I need to do, more comments etc.
I'm mainly interested to know if there is a logical fallacy in my basic idea, which somehow affects resource safety. If that's not the case, I'll do the necessary work to make this ready for merge :)

Reply:
Especially because we need to fix `interruptibleEval` asap, so it would be nicer to somehow linearise the changes to Scope to avoid making a mess :)